contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_present_in_other_table.py

```diff
@@ -0,0 +1,192 @@
+        LEFT JOIN {foreign_table} b
+            ON a.{foreign_key_column} = b.{primary_key_column_in_foreign_table}
+        WHERE b.{primary_key_column_in_foreign_table} is NULL)
+        """
+    success_keys = (
+        "template_dict",
+        "query",
+    )
+    domain_keys = ("query", "batch_id", "row_condition", "condition_parser")
+
+    default_kwarg_values = {
+        "catch_exceptions": False,
+        "meta": None,
+        "query": query,
+    }
+
+    def _validate_template_dict(
+        self, configuration: Optional[ExpectationConfiguration] = None
+    ) -> None:
+        template_dict = configuration.kwargs.get("template_dict")
+        if not isinstance(template_dict, dict):
+            raise TypeError("template_dict must be supplied as a dict")
+        if not all(
+            [
+                "foreign_key_column" in template_dict,
+                "foreign_table" in template_dict,
+                "primary_key_column_in_foreign_table" in template_dict,
+            ]
+        ):
+            raise KeyError(
+                "The following keys have to be in the template dict: id, foreign_table, foreign_key_column"
+            )
+
+    def _validate(
+        self,
+        metrics: dict,
+        runtime_configuration: Optional[dict] = None,
+        execution_engine: Optional[ExecutionEngine] = None,
+    ) -> Union[ExpectationValidationResult, dict]:
+        configuration = self.configuration
+        self._validate_template_dict(configuration)
+        as_dict_as_template = metrics.get("query.template_values")[0]
+        final_value = list(as_dict_as_template.values())[0]
+        return {
+            "success": final_value == 0,
+            "result": {
+                "Rows with IDs that are not present in foreign table": final_value
+            },
+        }
+
+    examples = [
+        {
+            "data": [
+                {
+                    "dataset_name": "order_table_1",
+                    "data": {
+                        "ORDER_ID": ["aaa", "bbb", "ccc"],
+                        "CUSTOMER_ID": [1, 1, 3],
+                    },
+                },
+                {
+                    "dataset_name": "customer_table",
+                    "data": {
+                        "CUSTOMER_ID": [1, 2, 3],
+                    },
+                },
+            ],
+            "only_for": ["snowflake", "sqlite"],
+            "tests": [
+                {
+                    "title": "basic_positive_test",
+                    "exact_match_out": False,
+                    "include_in_gallery": True,
+                    "in": {
+                        "template_dict": {
+                            "foreign_key_column": "CUSTOMER_ID",
+                            "foreign_table": "customer_table",
+                            "primary_key_column_in_foreign_table": "CUSTOMER_ID",
+                        }
+                    },
+                    "out": {"success": True},
```

Comment:
Do we typically test the "result" field as well? E.g. number of values in `final_value`? Could we?

Reply:
result has been updated to be : 
```python
"out": {"success": False, "unexpected_count": 2},
```